### PR TITLE
Adds withCount query constraint

### LINKED ISSUE #865: Add count flag to Parse.Query.find
REST api supports adding &count=true.
Using JS SDK one needs to run a separate query.count() to get this number

Would be nice to have some optional flag like query.withCount(true/false) to avoid
making separate async calls

### PR DESCRIPTION
**Pull request by BufferUnderflower**
Fixes #865

### CODE REVIEW

File: src/ParseQuery.js [hunk 1]
```diff
@@ -460,6 +477,10 @@ class ParseQuery {
       this._exclude = json.excludeKeys.split(",");
     }
 
+    if (json.count) {
+      this._count = json.count !== 0;
```
Comment (davimacedo) [inline src/ParseQuery.js]: Do you think that `this._count = json.count === 1;` would be better?

File: src/ParseQuery.js [hunk 2]
```diff
@@ -605,6 +627,14 @@ class ParseQuery {
 
         return ParseObject.fromJSON(data, !select);
       });
+
+      const count = response.count;
+
+      if(typeof count === "number"){
+        return {results, count};
```
Comment (dplewis) [inline src/ParseQuery.js]: Can you write a few integration tests to ensure you get the correct response?

File: src/ParseQuery.js [hunk 3]
```diff
@@ -1468,6 +1498,21 @@ class ParseQuery {
...
+   * Note that result of this query will be wrapped as an object with
+   *`results`: holding {ParseObject} array and `count`: integer holding total number
+   * @param {boolean} b false - disable, true - enable.
+   * @return {Parse.Query} Returns the query, so you can chain this call.
+   */
+  withCount(includeCount: boolean): ParseQuery {
+    if (typeof includeCount !== 'boolean') {
+      throw new Error('You can only set withCount to a boolean value');
+    }
+    this._count = includeCount;
```
Comment (dplewis) [inline src/ParseQuery.js]: Can you set a default?

`withCount(includeCount: boolean = true): ParseQuery {`

**Comment by davimacedo** [comment]
Good job @BufferUnderflower ! Would you be willed to add some information about it in JS docs?

**Comment by BufferUnderflower** [comment]
@davimacedo https://github.com/parse-community/docs/pull/644

**Pull request merged**